backend/src/main/java/ch/puzzle/okr/service/business/KeyResultBusinessService.java

```diff
@@ -137,4 +143,49 @@ private boolean isKeyResultTypeChangeable(Long id) {
     public List<KeyResult> getKeyResultsOwnedByUser(long id) {
         return keyResultPersistenceService.getKeyResultsOwnedByUser(id);
     }
+
+    @Transactional
+    public KeyResult duplicateKeyResult(AuthorizationUser authorizationUser, KeyResult keyResult,
+                                        Objective duplicatedObjective) {
+        KeyResult newKeyResult;
+
+        if (keyResult.getKeyResultType().equals(KEY_RESULT_TYPE_METRIC)) {
+            KeyResult keyResultMetric = makeCopyOfKeyResultMetric(keyResult, duplicatedObjective);
+            newKeyResult = createEntity(keyResultMetric, authorizationUser);
+        } else if (keyResult.getKeyResultType().equals(KEY_RESULT_TYPE_ORDINAL)) {
+            KeyResult keyResultOrdinal = makeCopyOfKeyResultOrdinal(keyResult, duplicatedObjective);
+            newKeyResult = createEntity(keyResultOrdinal, authorizationUser);
+        } else {
+            throw new UnsupportedOperationException("Unsupported KeyResultType: " + keyResult.getKeyResultType());
+        }
```

Comment:
```suggestion
                KeyResult newKeyResult = switch (keyResult.getKeyResultType()){
            case KEY_RESULT_TYPE_METRIC -> makeCopyOfKeyResultMetric(keyResult, duplicatedObjective);
            case KEY_RESULT_TYPE_ORDINAL -> makeCopyOfKeyResultOrdinal(keyResult, duplicatedObjective);
            default -> throw new UnsupportedOperationException("Unsupported KeyResultType: " + keyResult.getKeyResultType());
        };
        newKeyResult = createEntity(newKeyResult, authorizationUser);
        
```
This could be simplified somewhat by using the newish switch-statement instead of an if-else chain.